### ISSUE TITLE
Delay creation of hook until dll is loaded

### ISSUE THREAD

**Issue by paulnd** [EasyHook/EasyHook#241]
I’m trying to hook a dll using the C# example but the dll is only loaded when the API is used and then unloaded.

Does anyone have any suggestions please for delaying creating the hook until the dll is loaded or only while the dll is loaded?

Thanks

**Comment by duketwo** [comment]
LoadLibrary?

**Comment by spazzarama** [comment]
Yes you could try hooking loadlibrary and check if the target module is being loaded or not.

**Comment by paulnd** [comment]
Thanks, I'll give that a try

**Comment by fl380** [comment]
I've successfully hooked LoadLibrary and able to detect when the specific dll is loaded but how do I then create a hook into that dll?

There's a similar thread here [https://stackoverflow.com/a/50975505](url) but unfortunately it doesn't explain how to apply the second hook. I don't have enough reps on stockoverflow to add a comment there unfortunately.

I'll be grateful for any help please. thanks.

**Comment by spazzarama** [comment]
You could load the library yourself before installing local hook

**Comment by duketwo** [comment]
You also could hook FreeLibrary to prevent the DLL from being unloaded
